h2o-automl/src/test/java/ai/h2o/automl/AutoMLTest.java

```diff
@@ -192,6 +192,58 @@
     }
   }
 
+
+  @Test public void test_individual_model_max_runtime() {
+    AutoML aml=null;
+    Frame fr=null;
+    try {
+      AutoMLBuildSpec autoMLBuildSpec = new AutoMLBuildSpec();
+//      fr = parse_test_file("./smalldata/prostate/prostate_complete.csv"); //using slightly larger dataset to make this test useful
+//      autoMLBuildSpec.input_spec.response_column = "CAPSULE";
+      fr = parse_test_file("./smalldata/diabetes/diabetes_text_train.csv"); //using slightly larger dataset to make this test useful
+      autoMLBuildSpec.input_spec.response_column = "diabetesMed";
+      autoMLBuildSpec.input_spec.training_frame = fr._key;
+
+      int max_runtime_secs_per_model = 10;
+      autoMLBuildSpec.build_models.exclude_algos = new Algo[] {Algo.DRF};
+      autoMLBuildSpec.build_control.stopping_criteria.set_seed(1);
+      autoMLBuildSpec.build_control.stopping_criteria.set_max_models(3);
+      autoMLBuildSpec.build_control.stopping_criteria.set_max_runtime_secs_per_model(max_runtime_secs_per_model);
+      autoMLBuildSpec.build_control.keep_cross_validation_models = true; //Prevent leaked keys from CV models
+      autoMLBuildSpec.build_control.keep_cross_validation_predictions = false; //Prevent leaked keys from CV predictions
+
+      aml = AutoML.startAutoML(autoMLBuildSpec);
+      aml.get();
+
+      int tolerance = (autoMLBuildSpec.build_control.nfolds + 1) * 2; //generously adding 2s tolerance for each model
+      for (Key<Model> key : aml.leaderboard().getModelKeys()) {
+        Model model = key.get();
+        double duration = model._output._run_time;
+        Log.info(key + " build duration (s): "+ duration / 1e3);
+        if (model._output._cross_validation_models != null) {
+          for (Key<Model> kcv : model._output._cross_validation_models) {
+            Model cv = kcv.get();
+            long cv_duration = cv._output._run_time;
+            Log.info(kcv + " build duration (s): "+ cv_duration / 1e3);
+            duration += cv_duration;
+          }
+          Log.info(key + " added build duration (s): "+ duration / 1e3);
+        }
+        duration = model._output._total_run_time / 1e3;
+        Log.info(key + " model total build duration (s): "+ duration);
+        assertTrue(key + " took longer than required: "+ duration,
+            duration - max_runtime_secs_per_model < tolerance);
+        model.deleteCrossValidationModels();
+      }
+
+      // no assertion, we just want to check leaked keys
```

Comment:
@seb-h2o What for we are summing up `duration` here? I would expect assertion here that training for each model  took predefined time +- generously added 2s or at least whole autoML process finished within expected timebox.

Reply:
yes, only the last assertion is useful now for this test.
Was using the duration on CVs to investigate timing issues at some point, this looks confusing now, as well as those logs. Will clean up.